From eecc282093e2b9236b45ed0c19d523adf34e105c Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Mon, 16 Dec 2019 12:25:27 +0200
Subject: [PATCH 040/269] s32g: Skip PRAM and XRDC init

DDR is not initialized, so we can't safely execute this now.
Besides, this function will not be required in its current shape anyway.

Also, XRDC init code crashes for some reason not previously apparent on
VDK.

To Be Revisited.

Upstream-Status: Pending 

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-4277
---
 plat/s32g/s32g275_bl31.c | 3 +++
 plat/s32g/s32g_psci.c    | 6 ++++++
 2 files changed, 9 insertions(+)

diff --git a/plat/s32g/s32g275_bl31.c b/plat/s32g/s32g275_bl31.c
index 6884d9ece..db9b7f8af 100644
--- a/plat/s32g/s32g275_bl31.c
+++ b/plat/s32g/s32g275_bl31.c
@@ -190,6 +190,8 @@ void bl31_platform_setup(void)
  */
 void bl31_plat_runtime_setup(void)
 {
+	WARN("Skipping %s(); PRAM and XRDC are not working anymore.", __func__);
+#if 0
 	s32g_psci_move_to_pram();
 	/* If we enable XRDC, the functional simulator will screech to a halt;
 	 * until a fix is provided, we'll just skip it
@@ -197,6 +199,7 @@ void bl31_plat_runtime_setup(void)
 	INFO("Setting up XRDC...\n");
 	if (xrdc_enable((void *)S32G_XRDC_BASE))
 		ERROR("%s(): Error initializing XRDC!\n", __func__);
+#endif
 }
 
 unsigned int plat_get_syscnt_freq2(void)
diff --git a/plat/s32g/s32g_psci.c b/plat/s32g/s32g_psci.c
index 03bc3f4fe..1a58952f9 100644
--- a/plat/s32g/s32g_psci.c
+++ b/plat/s32g/s32g_psci.c
@@ -164,6 +164,11 @@ const plat_psci_ops_t s32g_psci_pm_ops = {
  */
 void s32g_psci_move_to_pram(void)
 {
+	/* DDR is not initialized, and besides we won't do relocation this way
+	 * anymore, but instead will seek to deploy BL31 entirely in DDR
+	 */
+	WARN("Skipping %s(); will need to revisit it.", __func__);
+#if 0
 	INFO("Copying self (0x%lx .. 0x%lx) to DRAM (0x%lx)...",
 	     bl31_start, bl31_end, (unsigned long)S32G_PMEM_START);
 	/* FIXME this may be too time-consuming; we should do it via DMA and/or
@@ -180,6 +185,7 @@ void s32g_psci_move_to_pram(void)
 	       (unsigned char *)bl31_start,
 	       bl31_end - bl31_start);
 	puts(" Done.");
+#endif
 }
 
 int plat_setup_psci_ops(uintptr_t sec_entrypoint,
-- 
2.17.1

